### PR TITLE
Orca: set conf to None when using spark-submit

### PR DESCRIPTION
**Pull request by lalalapotter**
## Description
This PR is to fix the issues report by customers when using spark-submit to submit tasks. If we use `conf` in `init_orca_context`, the `conf` will not be convert to `SparkConf` when using spark-submit and lead to exception. In this PR, we check the `conf`  before using it as `SparkConf` to avoid `no attribute 'getAll()'` exception.

### CODE REVIEW

File: python/orca/src/bigdl/orca/common.py [hunk 1]
```diff
@@ -250,6 +250,10 @@ def init_orca_context(cluster_mode=None, runtime="spark", cores=2, memory="2g",
                 sc = init_spark_on_local(cores, **spark_args)
             elif cluster_mode == "spark-submit":
                 from bigdl.dllib.nncontext import init_nncontext
+                if "conf" in spark_args and spark_args["conf"] is not None:
+                    warnings.warn("For Spark-submit mode, conf should be None, but got "
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/common.py:254]: Spark-submit mode -> spark-submit cluster_mode

Comment (hkvision) [reply python/orca/src/bigdl/orca/common.py:254]: change to `all conf should be specified in the spark-submit command`

**Comment by lalalapotter** [comment]
PR Validation Results:

[Passed](http://10.112.231.51:18888/job/BigDL-Orca-PR-Validation/868/), several test failed at first validation and passed when rerunning the test.

File: python/orca/src/bigdl/orca/common.py (hunk 1 above)
Comment (lalalapotter) [reply python/orca/src/bigdl/orca/common.py]: fixed in new commit

**Review by hkvision — APPROVED** [review]

File: python/orca/src/bigdl/orca/common.py [hunk 2]
```diff
@@ -250,6 +250,11 @@ def init_orca_context(cluster_mode=None, runtime="spark", cores=2, memory="2g",
                 sc = init_spark_on_local(cores, **spark_args)
             elif cluster_mode == "spark-submit":
                 from bigdl.dllib.nncontext import init_nncontext
+                if "conf" in spark_args and spark_args["conf"] is not None:
+                    warnings.warn("For spark-submit cluster_mode, all conf should be "
+                                  + "specified in the spark-submit command , but got "
```
[the remark this answers is not in the document]
Comment (lalalapotter) [reply python/orca/src/bigdl/orca/common.py]: fixed

**Pull request merged**